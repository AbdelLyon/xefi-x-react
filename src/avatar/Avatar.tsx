import type { JSX } from "react";
import { forwardRef } from "react";
import type { AvatarGroupProps, AvatarProps, UserProps } from "@heroui/react";
import {
  Avatar as AvatarRoot,
  User as UserRoot,
  useAvatarGroup,
  AvatarGroupProvider,
} from "@heroui/react";

export const Avatar = forwardRef<HTMLDivElement, AvatarProps>(
  (props, ref): JSX.Element => {
    return <AvatarRoot ref={ref} {...props} />;
  },
);

Avatar.displayName = "Avatar";

export const AvatarGroup = forwardRef<HTMLDivElement, AvatarGroupProps>(
  (props, ref): JSX.Element => {
    const {
      Component,
      clones,
      context,
      remainingCount,
      renderCount = (count): JSX.Element => <Avatar name={`+${count}`} />,
      getAvatarGroupProps,
    } = useAvatarGroup({
      ref,
      ...props,
    });

    return (
      <Component {...getAvatarGroupProps()}>
        <AvatarGroupProvider value={context}>
          {clones}
          {remainingCount > 0 && renderCount(remainingCount)}
        </AvatarGroupProvider>
      </Component>
    );
  },
);

AvatarGroup.displayName = "AvatarGroup";

export const UserAvatar = forwardRef<HTMLDivElement, UserProps>(
  (props, ref): JSX.Element => {
    return <UserRoot ref={ref} {...props} />;
  },
);

UserAvatar.displayName = "UserAvatar";

Avatar.displayName = "Avatar";
