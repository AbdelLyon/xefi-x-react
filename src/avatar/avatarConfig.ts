/**
 * Avatar component configuration
 * Provides standardized styling and behavior patterns
 */

import { createClassNamesConfig, type ComponentClassNames } from "@/utils/classNames";
import type { ReactNode } from "react";

/**
 * Default class names for avatar components
 */
export const defaultAvatarClassNames = {
  base: "flex-shrink-0",
  img: "object-cover",
  fallback: "font-normal text-center uppercase",
  name: "text-inherit",
  icon: "text-inherit",
} as const;

/**
 * Avatar size class configurations
 */
export const avatarSizeClasses = {
  sm: {
    base: "w-8 h-8 text-xs",
    name: "text-xs",
  },
  md: {
    base: "w-10 h-10 text-sm", 
    name: "text-sm",
  },
  lg: {
    base: "w-12 h-12 text-base",
    name: "text-base",
  },
  xl: {
    base: "w-16 h-16 text-lg",
    name: "text-lg",
  },
} as const;

/**
 * Avatar color class configurations
 */
export const avatarColorClasses = {
  default: {
    base: "bg-default text-default-foreground",
  },
  primary: {
    base: "bg-primary text-primary-foreground", 
  },
  secondary: {
    base: "bg-secondary text-secondary-foreground",
  },
  success: {
    base: "bg-success text-success-foreground",
  },
  warning: {
    base: "bg-warning text-warning-foreground",
  },
  danger: {
    base: "bg-danger text-danger-foreground",
  },
} as const;

/**
 * Avatar status indicator classes
 */
export const avatarStatusClasses = {
  online: "bg-success border-2 border-white dark:border-background",
  offline: "bg-default-300 border-2 border-white dark:border-background",
  away: "bg-warning border-2 border-white dark:border-background",
  busy: "bg-danger border-2 border-white dark:border-background",
} as const;

/**
 * Create avatar class configuration
 */
export const avatarClassConfig = createClassNamesConfig(defaultAvatarClassNames);

/**
 * Get avatar size classes
 */
export const getAvatarSizeClasses = (size: keyof typeof avatarSizeClasses = "md") => {
  return avatarSizeClasses[size] || avatarSizeClasses.md;
};

/**
 * Get avatar color classes  
 */
export const getAvatarColorClasses = (color: keyof typeof avatarColorClasses = "default") => {
  return avatarColorClasses[color] || avatarColorClasses.default;
};

/**
 * Avatar configuration interface
 */
export interface AvatarConfig {
  /** Avatar image source */
  src?: string;
  /** Alternative text for image */
  alt?: string;
  /** Name to display as fallback */
  name?: string;
  /** Icon to display as fallback */
  icon?: ReactNode;
  /** Avatar size */
  size?: keyof typeof avatarSizeClasses;
  /** Avatar color theme */
  color?: keyof typeof avatarColorClasses;
  /** Avatar shape */
  radius?: "none" | "sm" | "md" | "lg" | "full";
  /** Whether avatar is bordered */
  isBordered?: boolean;
  /** Whether avatar is disabled */
  isDisabled?: boolean;
  /** Status indicator */
  status?: keyof typeof avatarStatusClasses;
  /** Custom class names */
  classNames?: ComponentClassNames<typeof defaultAvatarClassNames>;
}

/**
 * Avatar group configuration
 */
export interface AvatarGroupConfig {
  /** Maximum avatars to show */
  max?: number;
  /** Avatar size for all items */
  size?: keyof typeof avatarSizeClasses;
  /** Whether avatars are bordered */
  isBordered?: boolean;
  /** Whether group is disabled */
  isDisabled?: boolean;
  /** Custom render function for overflow count */
  renderCount?: (count: number) => ReactNode;
  /** Grid layout for avatars */
  isGrid?: boolean;
  /** Spacing between avatars */
  spacing?: "sm" | "md" | "lg";
}

/**
 * Generate initials from name
 */
export const generateInitials = (name: string, maxLength: number = 2): string => {
  if (!name || typeof name !== "string") {return "";}
  
  const words = name.trim().split(/\\s+/);
  
  if (words.length === 1) {
    return words[0].substring(0, maxLength).toUpperCase();
  }
  
  return words
    .slice(0, maxLength) 
    .map(word => word.charAt(0))
    .join("")
    .toUpperCase();
};

/**
 * Validate avatar configuration
 */
export const validateAvatarConfig = (config: AvatarConfig): { valid: boolean; errors: string[] } => {
  const errors: string[] = [];
  
  // Must have at least one display method
  if (!config.src && !config.name && !config.icon) {
    errors.push("Avatar must have either src, name, or icon");
  }
  
  // Validate image URL if provided
  if (config.src && typeof config.src !== "string") {
    errors.push("Avatar src must be a valid string URL");
  }
  
  // Validate name if provided
  if (config.name && typeof config.name !== "string") {
    errors.push("Avatar name must be a string");
  }
  
  return {
    valid: errors.length === 0,
    errors,
  };
};

/**
 * Default avatar behavior configuration
 */
export const defaultAvatarBehavior = {
  size: "md" as const,
  color: "default" as const,
  radius: "full" as const,
  isBordered: false,
  isDisabled: false,
  showFallback: true,
  fallbackSrc: undefined,
} as const;

/**
 * Create avatar props with defaults
 */
export const createAvatarProps = <T extends Record<string, unknown>>(
  userProps: T,
  overrides?: Partial<typeof defaultAvatarBehavior>
) => {
  return {
    ...defaultAvatarBehavior,
    ...overrides,
    ...userProps,
  };
};