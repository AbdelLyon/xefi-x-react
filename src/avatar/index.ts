import "@/index.css";
export { Avatar, AvatarGroup, UserAvatar } from "@/avatar/Avatar";
