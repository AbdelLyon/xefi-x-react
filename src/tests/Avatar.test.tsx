import { Avatar, AvatarGroup, UserAvatar } from "@/avatar";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";

describe("Composants Avatar", (): void => {
  describe("Avatar", (): void => {
    describe("Snapshots", (): void => {
      it("devrait correspondre au snapshot sans props", (): void => {
        const { container } = render(<Avatar />);
        expect(container).toMatchSnapshot();
      });

      it("devrait correspondre au snapshot avec src et alt", (): void => {
        const { container } = render(
          <Avatar src="test.jpg" alt="Mon avatar" />,
        );
        expect(container).toMatchSnapshot();
      });

      it("devrait correspondre au snapshot avec name", (): void => {
        const { container } = render(<Avatar name="John Doe" />);
        expect(container).toMatchSnapshot();
      });

      it("devrait correspondre au snapshot avec taille différente", (): void => {
        const { container } = render(<Avatar size="lg" />);
        expect(container).toMatchSnapshot();
      });

      it("devrait correspondre au snapshot avec fallback", (): void => {
        const { container } = render(
          <Avatar
            src="invalid.jpg"
            fallback={<div data-testid="fallback">FB</div>}
            showFallback
          />,
        );
        expect(container).toMatchSnapshot();
      });
    });
  });

  describe("AvatarGroup", (): void => {
    describe("Snapshots", (): void => {
      it("devrait correspondre au snapshot pour un groupe simple", (): void => {
        const { container } = render(
          <AvatarGroup>
            <Avatar name="User 1" />
            <Avatar name="User 2" />
          </AvatarGroup>,
        );
        expect(container).toMatchSnapshot();
      });

      it("devrait correspondre au snapshot avec max et total", (): void => {
        const { container } = render(
          <AvatarGroup max={1} total={3}>
            <Avatar name="User 1" />
            <Avatar name="User 2" />
            <Avatar name="User 3" />
          </AvatarGroup>,
        );
        expect(container).toMatchSnapshot();
      });

      it("devrait correspondre au snapshot avec styles personnalisés", (): void => {
        const { container } = render(
          <AvatarGroup className="size-6">
            <Avatar />
          </AvatarGroup>,
        );
        expect(container).toMatchSnapshot();
      });
    });
  });

  describe("UserAvatar", (): void => {
    describe("Snapshots", (): void => {
      it("devrait correspondre au snapshot avec informations de base", (): void => {
        const { container } = render(
          <UserAvatar name="John Doe" description="Developer" />,
        );
        expect(container).toMatchSnapshot();
      });

      it("devrait correspondre au snapshot avec avatar personnalisé", (): void => {
        const { container } = render(
          <UserAvatar
            name="John Doe"
            avatarProps={{
              src: "test.jpg",
              alt: "John",
            }}
          />,
        );
        expect(container).toMatchSnapshot();
      });

      it("devrait correspondre au snapshot avec contenus personnalisés", (): void => {
        const { container } = render(
          <UserAvatar
            name={<span data-testid="custom-name">John</span>}
            description={<span data-testid="custom-desc">Info</span>}
          />,
        );
        expect(container).toMatchSnapshot();
      });
    });
  });
});
