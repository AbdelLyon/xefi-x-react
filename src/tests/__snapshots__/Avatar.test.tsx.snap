// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Composants Avatar > Avatar > Snapshots > devrait correspondre au snapshot avec fallback 1`] = `
<div>
  <span
    class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full"
    tabindex="-1"
  >
    <img
      alt="avatar"
      class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
      src="invalid.jpg"
    />
    <div
      aria-label="avatar"
      class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 flex items-center justify-center"
      role="img"
    >
      <div
        data-testid="fallback"
      >
        FB
      </div>
    </div>
  </span>
</div>
`;

exports[`Composants Avatar > Avatar > Snapshots > devrait correspondre au snapshot avec name 1`] = `
<div>
  <span
    class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full"
    tabindex="-1"
  >
    <span
      aria-label="John Doe"
      class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 font-normal text-center text-inherit"
      role="img"
    >
      Joh
    </span>
  </span>
</div>
`;

exports[`Composants Avatar > Avatar > Snapshots > devrait correspondre au snapshot avec src et alt 1`] = `
<div>
  <span
    class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full"
    tabindex="-1"
  >
    <img
      alt="Mon avatar"
      class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
      src="test.jpg"
    />
  </span>
</div>
`;

exports[`Composants Avatar > Avatar > Snapshots > devrait correspondre au snapshot avec taille différente 1`] = `
<div>
  <span
    class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-14 h-14 text-small bg-default text-default-foreground rounded-full"
    tabindex="-1"
  >
    <span
      aria-label="avatar"
      class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 flex items-center justify-center text-inherit w-full h-full"
      role="img"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="80%"
        role="presentation"
        viewBox="0 0 24 24"
        width="80%"
      >
        <path
          d="M12 2C9.38 2 7.25 4.13 7.25 6.75C7.25 9.32 9.26 11.4 11.88 11.49C11.96 11.48 12.04 11.48 12.1 11.49C12.12 11.49 12.13 11.49 12.15 11.49C12.16 11.49 12.16 11.49 12.17 11.49C14.73 11.4 16.74 9.32 16.75 6.75C16.75 4.13 14.62 2 12 2Z"
          fill="currentColor"
        />
        <path
          d="M17.0809 14.1489C14.2909 12.2889 9.74094 12.2889 6.93094 14.1489C5.66094 14.9989 4.96094 16.1489 4.96094 17.3789C4.96094 18.6089 5.66094 19.7489 6.92094 20.5889C8.32094 21.5289 10.1609 21.9989 12.0009 21.9989C13.8409 21.9989 15.6809 21.5289 17.0809 20.5889C18.3409 19.7389 19.0409 18.5989 19.0409 17.3589C19.0309 16.1289 18.3409 14.9889 17.0809 14.1489Z"
          fill="currentColor"
        />
      </svg>
    </span>
  </span>
</div>
`;

exports[`Composants Avatar > Avatar > Snapshots > devrait correspondre au snapshot sans props 1`] = `
<div>
  <span
    class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full"
    tabindex="-1"
  >
    <span
      aria-label="avatar"
      class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 flex items-center justify-center text-inherit w-full h-full"
      role="img"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="80%"
        role="presentation"
        viewBox="0 0 24 24"
        width="80%"
      >
        <path
          d="M12 2C9.38 2 7.25 4.13 7.25 6.75C7.25 9.32 9.26 11.4 11.88 11.49C11.96 11.48 12.04 11.48 12.1 11.49C12.12 11.49 12.13 11.49 12.15 11.49C12.16 11.49 12.16 11.49 12.17 11.49C14.73 11.4 16.74 9.32 16.75 6.75C16.75 4.13 14.62 2 12 2Z"
          fill="currentColor"
        />
        <path
          d="M17.0809 14.1489C14.2909 12.2889 9.74094 12.2889 6.93094 14.1489C5.66094 14.9989 4.96094 16.1489 4.96094 17.3789C4.96094 18.6089 5.66094 19.7489 6.92094 20.5889C8.32094 21.5289 10.1609 21.9989 12.0009 21.9989C13.8409 21.9989 15.6809 21.5289 17.0809 20.5889C18.3409 19.7389 19.0409 18.5989 19.0409 17.3589C19.0309 16.1289 18.3409 14.9889 17.0809 14.1489Z"
          fill="currentColor"
        />
      </svg>
    </span>
  </span>
</div>
`;

exports[`Composants Avatar > AvatarGroup > Snapshots > devrait correspondre au snapshot avec max et total 1`] = `
<div>
  <div
    class="flex items-center justify-center h-auto w-max"
    role="group"
  >
    <span
      class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ms-0"
      tabindex="-1"
    >
      <span
        aria-label="User 1"
        class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 font-normal text-center text-inherit"
        role="img"
      >
        Use
      </span>
    </span>
    <span
      class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full -ms-2 data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3"
      tabindex="-1"
    >
      <span
        aria-label="+3"
        class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 font-normal text-center text-inherit"
        role="img"
      >
        +3
      </span>
    </span>
  </div>
</div>
`;

exports[`Composants Avatar > AvatarGroup > Snapshots > devrait correspondre au snapshot avec styles personnalisés 1`] = `
<div>
  <div
    class="flex items-center justify-center size-6"
    role="group"
  >
    <span
      class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ms-0 hover:-translate-x-0"
      tabindex="-1"
    >
      <span
        aria-label="avatar"
        class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 flex items-center justify-center text-inherit w-full h-full"
        role="img"
      >
        <svg
          aria-hidden="true"
          fill="none"
          height="80%"
          role="presentation"
          viewBox="0 0 24 24"
          width="80%"
        >
          <path
            d="M12 2C9.38 2 7.25 4.13 7.25 6.75C7.25 9.32 9.26 11.4 11.88 11.49C11.96 11.48 12.04 11.48 12.1 11.49C12.12 11.49 12.13 11.49 12.15 11.49C12.16 11.49 12.16 11.49 12.17 11.49C14.73 11.4 16.74 9.32 16.75 6.75C16.75 4.13 14.62 2 12 2Z"
            fill="currentColor"
          />
          <path
            d="M17.0809 14.1489C14.2909 12.2889 9.74094 12.2889 6.93094 14.1489C5.66094 14.9989 4.96094 16.1489 4.96094 17.3789C4.96094 18.6089 5.66094 19.7489 6.92094 20.5889C8.32094 21.5289 10.1609 21.9989 12.0009 21.9989C13.8409 21.9989 15.6809 21.5289 17.0809 20.5889C18.3409 19.7389 19.0409 18.5989 19.0409 17.3589C19.0309 16.1289 18.3409 14.9889 17.0809 14.1489Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </span>
  </div>
</div>
`;

exports[`Composants Avatar > AvatarGroup > Snapshots > devrait correspondre au snapshot pour un groupe simple 1`] = `
<div>
  <div
    class="flex items-center justify-center h-auto w-max"
    role="group"
  >
    <span
      class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 ms-0"
      tabindex="-1"
    >
      <span
        aria-label="User 1"
        class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 font-normal text-center text-inherit"
        role="img"
      >
        Use
      </span>
    </span>
    <span
      class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full data-[hover=true]:-translate-x-3 rtl:data-[hover=true]:translate-x-3 transition-transform data-[focus-visible=true]:-translate-x-3 rtl:data-[focus-visible=true]:translate-x-3 -ms-2 hover:-translate-x-0"
      tabindex="-1"
    >
      <span
        aria-label="User 2"
        class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 font-normal text-center text-inherit"
        role="img"
      >
        Use
      </span>
    </span>
  </div>
</div>
`;

exports[`Composants Avatar > UserAvatar > Snapshots > devrait correspondre au snapshot avec avatar personnalisé 1`] = `
<div>
  <div
    class="inline-flex items-center justify-center gap-2 rounded-small outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
    tabindex="-1"
  >
    <span
      class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full"
      tabindex="-1"
    >
      <img
        alt="John"
        class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
        src="test.jpg"
      />
    </span>
    <div
      class="inline-flex flex-col items-start"
    >
      <span
        class="text-small text-inherit"
      >
        John Doe
      </span>
      <span
        class="text-tiny text-foreground-400"
      />
    </div>
  </div>
</div>
`;

exports[`Composants Avatar > UserAvatar > Snapshots > devrait correspondre au snapshot avec contenus personnalisés 1`] = `
<div>
  <div
    class="inline-flex items-center justify-center gap-2 rounded-small outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
    tabindex="-1"
  >
    <span
      class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full"
      tabindex="-1"
    >
      <span
        aria-label="avatar"
        class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 flex items-center justify-center text-inherit w-full h-full"
        role="img"
      >
        <svg
          aria-hidden="true"
          fill="none"
          height="80%"
          role="presentation"
          viewBox="0 0 24 24"
          width="80%"
        >
          <path
            d="M12 2C9.38 2 7.25 4.13 7.25 6.75C7.25 9.32 9.26 11.4 11.88 11.49C11.96 11.48 12.04 11.48 12.1 11.49C12.12 11.49 12.13 11.49 12.15 11.49C12.16 11.49 12.16 11.49 12.17 11.49C14.73 11.4 16.74 9.32 16.75 6.75C16.75 4.13 14.62 2 12 2Z"
            fill="currentColor"
          />
          <path
            d="M17.0809 14.1489C14.2909 12.2889 9.74094 12.2889 6.93094 14.1489C5.66094 14.9989 4.96094 16.1489 4.96094 17.3789C4.96094 18.6089 5.66094 19.7489 6.92094 20.5889C8.32094 21.5289 10.1609 21.9989 12.0009 21.9989C13.8409 21.9989 15.6809 21.5289 17.0809 20.5889C18.3409 19.7389 19.0409 18.5989 19.0409 17.3589C19.0309 16.1289 18.3409 14.9889 17.0809 14.1489Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </span>
    <div
      class="inline-flex flex-col items-start"
    >
      <span
        class="text-small text-inherit"
      >
        <span
          data-testid="custom-name"
        >
          John
        </span>
      </span>
      <span
        class="text-tiny text-foreground-400"
      >
        <span
          data-testid="custom-desc"
        >
          Info
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`Composants Avatar > UserAvatar > Snapshots > devrait correspondre au snapshot avec informations de base 1`] = `
<div>
  <div
    class="inline-flex items-center justify-center gap-2 rounded-small outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
    tabindex="-1"
  >
    <span
      class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full"
      tabindex="-1"
    >
      <span
        aria-label="avatar"
        class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 flex items-center justify-center text-inherit w-full h-full"
        role="img"
      >
        <svg
          aria-hidden="true"
          fill="none"
          height="80%"
          role="presentation"
          viewBox="0 0 24 24"
          width="80%"
        >
          <path
            d="M12 2C9.38 2 7.25 4.13 7.25 6.75C7.25 9.32 9.26 11.4 11.88 11.49C11.96 11.48 12.04 11.48 12.1 11.49C12.12 11.49 12.13 11.49 12.15 11.49C12.16 11.49 12.16 11.49 12.17 11.49C14.73 11.4 16.74 9.32 16.75 6.75C16.75 4.13 14.62 2 12 2Z"
            fill="currentColor"
          />
          <path
            d="M17.0809 14.1489C14.2909 12.2889 9.74094 12.2889 6.93094 14.1489C5.66094 14.9989 4.96094 16.1489 4.96094 17.3789C4.96094 18.6089 5.66094 19.7489 6.92094 20.5889C8.32094 21.5289 10.1609 21.9989 12.0009 21.9989C13.8409 21.9989 15.6809 21.5289 17.0809 20.5889C18.3409 19.7389 19.0409 18.5989 19.0409 17.3589C19.0309 16.1289 18.3409 14.9889 17.0809 14.1489Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </span>
    <div
      class="inline-flex flex-col items-start"
    >
      <span
        class="text-small text-inherit"
      >
        John Doe
      </span>
      <span
        class="text-tiny text-foreground-400"
      >
        Developer
      </span>
    </div>
  </div>
</div>
`;
